Only save history if Vsaving_history is true.  Thanks to John Eaton
for the patch and the comments.  Closes Bug#514802.

 -- Rafael Laboissiere <rafael@debian.org>  Fri, 13 Feb 2009 19:52:37 +0100

--- octave3.0-3.0.4~rc3.orig/src/toplev.cc
+++ octave3.0-3.0.4~rc3/src/toplev.cc
@@ -643,7 +643,8 @@
 
       octave_history_write_timestamp ();
 
-      command_history::clean_up_and_save ();
+      if (Vsaving_history)
+	command_history::clean_up_and_save ();
 
       close_files ();
 
--- octave3.0-3.0.4~rc3.orig/src/oct-hist.cc
+++ octave3.0-3.0.4~rc3/src/oct-hist.cc
@@ -534,7 +534,8 @@
   command_history::set_file (Vhistory_file);
   command_history::set_size (Vhistory_size);
 
-  command_history::read (false);
+  if (Vsaving_history)
+    command_history::read (false);
 }
 
 void
